src/Compilers/CSharp/Test/Symbol/Compilation/CompilationAPITests.cs

```diff
@@ -2053,6 +2053,30 @@ public void PreviousSubmissionWithError()
             Assert.Throws<InvalidOperationException>(() => CreateSubmission("a + 1", previous: s0));
         }
 
+        [Fact]
+        public void CreateTupleTypeSymbol()
+        {
+            var comp = CSharpCompilation.Create("test", references: new[] { MscorlibRef } );
+            ITypeSymbol intType = comp.GetSpecialType(SpecialType.System_Int32);
+            ITypeSymbol stringType = comp.GetSpecialType(SpecialType.System_String);
+
+            Assert.Throws<ArgumentNullException>(() => comp.CreateTupleTypeSymbol(default(ImmutableArray<ITypeSymbol>), default(ImmutableArray<string>)));
+
+            Assert.Throws<ArgumentException>(() => comp.CreateTupleTypeSymbol(ImmutableArray<ITypeSymbol>.Empty, default(ImmutableArray<string>)));
+            Assert.Throws<ArgumentException>(() => comp.CreateTupleTypeSymbol(new[] { intType }.AsImmutable(), default(ImmutableArray<string>)));
```

Comment:
We should likely also have a test for the case of 8 and 9 element tuples.  They are interesting cases for the language hence it reasons they should be interesting for the API as well.